.github/workflows/codeql-analysis.yml

```
@@ -0,0 +1,52 @@
+name: "Code scanning - action"
+
+on:
+  push:
```

I think it's best we run on master merges and PRs
```suggestion
  push:
    branches:
      - master
```